proc-macros/src/attributes.rs

```diff
@@ -189,3 +195,11 @@ where
 {
 	arg.ok().map(transform).transpose()
 }
+
+pub(crate) fn parse_param_kind(arg: Result<Argument, MissingArgument>) -> ParamKind {
+	match optional(arg, Argument::string).unwrap().as_deref() {
+		None | Some("array") => ParamKind::Array,
+		Some("map") => ParamKind::Map,
+		err => panic!("param_kind must be either `map` or `array`, got {:?}", err),
+	}
+}
```

Comment:
We should return an error here, unwrap is going to throw away span information on invalid parse of the argument value. Instead of a string we could parse to raw `Argument::value` using `syn::Ident` so you can skip quotes, plus then you have the span to use to emit an error instead of a panic.

No guarantees that this compiles:

```suggestion
pub(crate) fn parse_param_kind(arg: Result<Argument, MissingArgument>) -> syn::Result<ParamKind> {
	let kind: Option<syn::Ident> = optional(arg, Argument::value)?;

	match kind {
		None => Ok(ParamKind::Array),
		Some(ident) if ident == "array" => Ok(ParamKind::Array),
		Some(ident) if ident == "map" => Ok(ParamKind::Map),
		Some(ident) => Err(Error::new(ident.span(), "param_kind must be either `map` or `array`")),
	}
}
```

Reply:
I figure that this won't be necessary if we go with your suggestion of turning `parma_kind` into a flag. Personally, I prefer the more verbose version (that's why I've implemented it in this way), but I complete get your point. Since this is obviously not my library, I'm happy to turn it into a flag if you want, I just care about the feature as a whole.

Reply:
Yeah, this is only if we want to keep the verbose variant.

I don't have a super strong opinion on this, and don't necessarily want to turn it into a bike shed, and we can still change the syntax of it later before 0.5 is stabilized.